### DISCUSSION TITLE
Preferred approach for depending on a `StreamProvider`

### DISCUSSION THREAD

**Discussion by plammens** [rrousselGit/riverpod#2052] [Q&A] [answered]
When a provider depends on the value exposed by another `StreamProvider` (let's call it `parentProvider`), which of the following is the preferred approach?

1. Using `parentProvider.future`
```dart
final dependantProvider = FutureProvider((ref) async {
  final value = await ref.watch(parentProvider.future);
  return doSomethingWithValue(value);
})
```

2. Using `parentProvider.stream`
```dart
final dependantProvider = StreamProvider((ref) {
  return ref.watch(parentProvider.stream).map(doSomethingWithValue);
})
```

First of all, are these two approaches functionally equivalent?
I think that both will be used in the same way as both will expose an `AsyncValue`, is this correct?

If they are functionally equivalent, I guess the other aspect to think about is which makes more sense semantically. If a provider bases its exposed value on a stream's value, it makes sense that it also builds a stream, since it is basically transforming the original stream. But perhaps option 1. has more intuitive syntax... 

Any opinions?

--------------------

Addendum:
If the dependant provider itself needs to expose an asynchronous stream of values for each value of the parent, then the options would become

1. Using `parentProvider.future`
```dart
final dependantProvider = StreamProvider((ref) async* {
  final value = await ref.watch(parentProvider.future);
  yield* buildStream(value);
})
```

2. Using `parentProvider.stream`
```dart
final dependantProvider = StreamProvider((ref) {
  return ref.watch(parentProvider.stream).asyncExpand(buildStream);
})
```

**Comment by rrousselGit** [comment, marked as answer]
Using `.stream`is not recommended.
I'm probably going to remove it at some point

Using `.future` is fine
